File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public int[] solution(int[] arr, int[] delete_list) 
    {
        List<int> arrList = new List<int>(arr);
        
        foreach(int num in delete_list)
        {
            for(int i=0; i < arrList.Count; i++)
            {
                if(arrList[i] == num)
                {
                    arrList.Remove(num);
                }
            }
        }
        return arrList.ToArray();
    }
}

File: 프로그래머스/0/181844. 배열의 원소 삭제하기/배열의 원소 삭제하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<int> delete_list) 
{
    int i = 0;
    while(i < arr.size())
    {   
        bool isCheck = false;
        for(int &num : delete_list)
        {
            if(arr[i] == num)
            {
                arr.erase(arr.begin()+i);
                isCheck = true;
            }
        }
        if(isCheck)
        {
            i = 0;
        }
        else
        {
            i++;
        }
    }
    return arr;
}